frontend/src/machines/billingMachine.ts

```diff
@@ -1,3 +1,134 @@
-import { createMachine } from "xstate";
+import { createMachine, assign } from "xstate";
 
-export const billingMachine = createMachine({});
+type Billing = {
+  id: string;
+  title: string;
+  amount: number;
+};
+
+interface Context {
+  billings: Billing[];
+  formMode: "CREATE" | "EDIT";
+}
+
+const billingMachine = createMachine<Context>({
+  id: "billing",
+  initial: "idle",
+  context: {
+    billings: [],
+    formMode: "CREATE",
+  },
+  states: {
+    idle: {
+      on: {
+        FETCH_BILLING: {
+          target: "loadingBilling",
+        },
+      },
+    },
+    loadingBilling: {
+      on: {
+        FETCH_BILLING_SUCCES: {
+          target: "getBillingOK",
+        },
+        FETCH_BILLING_ERROR: {
+          target: "getBillingError",
+        },
+      },
+    },
+    getBillingOK: {
+      on: {
+        ACTIVATE_BILLING_FORM: {
+          target: "billingFormIdle",
+        },
+      },
+    },
+    getBillingError: {
+      on: {
+        REFETCH_BILLING: {
+          target: "loadingBilling",
+        },
+      },
+    },
+    billingFormIdle: {
+      always: [
+        {
+          cond: "creatingForm",
+          target: "billingFormReady",
+        },
+        {
+          target: "loadingBillingDetail",
+        },
+      ],
+    },
+    loadingBillingDetail: {
+      on: {
+        FETCHING_BILLING_DETAIL_SUCCES: {
+          target: "billingFormReady",
+        },
+        FETCHING_BILLING_DETAIL_ERROR: {
+          target: "getBillingDetailError",
+        },
+      },
+    },
+    getBillingDetailError: {
+      on: {
+        REFTECH_BILLING_DETAIL: {
+          target: "loadingBillingDetail",
+        },
+      },
+    },
+    billingFormReady: {
+      initial: "firstStep",
+      states: {
+        firstStep: {
+          id: "firstStep",
+          on: {
+            NEXT_STEP: {
+              target: "secondStep",
+            },
+          },
+        },
+        secondStep: {
+          id: "secondStep",
+          on: {
+            PREV_STEP: {
+              target: "firstStep",
+            },
+            SUBMIT_BILLING: {
+              target: "#submitBilling",
```

Comment:
ini kayaknya bisa langsung target state "submitBilling" aja instead of id ?

Reply:
Gabisa ka, karena si State `secondStep` ini tuh beda level sama state `submitBilling` nya.
In this case, kita perlu tetap pake `#` buat navigate dan provide ID di targeted state nya